typescript-codegen.ts

```diff
@@ -0,0 +1,241 @@
+    
+        // We could add multiple schemas for multiple types,
+        // but here we're just making one type from JSON schema.
+        await schemaInput.addSource({ name: typeName, schema: jsonSchemaString });
+    
+        const inputData = new InputData();
+        inputData.addInput(schemaInput);
+    
+        return await quicktype({
+            inputData,
+            lang: targetLanguage,
+        });
+    }
+
+    private generateDelete = (resourceName: string) => {
+        return `\tDelete = () : void => {
+    \t\tthis.HttpDelete(${resourceName.toUpperCase()}_ROUTE);
+    \t}${EOL}${EOL}`;
+    };
+    
+    private generatePatch = (resourceName: string) => {
+        return `\tModify = () : void => {
+    \t\tthis.HttpPatch(${resourceName.toUpperCase()}_ROUTE);
+    \t}${EOL}${EOL}`;
+    };
+    
+    private capitalizeFirstLetter = (string: string) => {
+        return string.charAt(0).toUpperCase() + string.slice(1);
+    };
+
+    public LoadPrimitives = async (primitivesPath: string) => {
+        const primitiveFileNames: Array<string> =
+            fs.readdirSync(primitivesPath);
+        primitiveFileNames.forEach(async (fileName) => {
+            const primitiveName = fileName.split(".")[0];
+            const fileContent = fs.readFileSync(
+                `${primitivesPath}/${fileName}`,
+                "utf8"
+            );
+            const typescriptInterfaceContent = await this.quicktypeJSONSchema(
+                "typescript",
+                primitiveName,
+                fileContent
+            );
+            this.primitives.push(
+                new PrimitiveData(
+                    fileName,
+                    fileContent,
+                    typescriptInterfaceContent
+                )
+            );
+        });
+        this.primitivesPath = primitivesPath;
+    };
+
+    public LoadResources = async (resourcesPath: string) => {
+        const resourcesFileNames: Array<string> = fs.readdirSync(resourcesPath);
+        resourcesFileNames.forEach(async (fileName) => {
+            const resourceName = fileName.split(".")[0];
+            const fileContent = fs.readFileSync(
+                `${resourcesPath}/${fileName}`,
+                "utf8"
+            );
+            const jsonContent = JSON.parse(fileContent);
+            const typescriptInterfaceContent = await this.quicktypeJSONSchema(
+                "typescript",
+                resourceName,
+                fileContent
+            );
+            const resourceRoute = this.getRelativePath(jsonContent);
+            this.resources.push(
+                new ResourceData(
+                    fileName,
+                    resourceRoute,
+                    fileContent,
+                    jsonContent,
+                    typescriptInterfaceContent
+                )
+            );
+        });
+        this.resourcesPath = resourcesPath;
+    };
+
+    public GenerateResources = async (generationPath: string) => {
```

Comment:
You're running async callbacks, that are just thrownn at the js event loop and executed on worker threads asynchronously and independedly, so concurrently.

Do you intend to sync them in some way?

Reply:
There is no need to mark this func as async. This is just the product of some blind shots to overcome the issue in the bottom of the file.